bfdd/ptm_adapter.c

```diff
@@ -204,12 +204,15 @@ int ptm_bfd_notify(struct bfd_session *bs)
 	_ptm_msg_address(msg, bs->key.family, &bs->key.peer);
 
 	/* BFD status */
-	switch (bs->ses_state) {
+	switch (notify_state) {
 	case PTM_BFD_UP:
 		stream_putl(msg, BFD_STATUS_UP);
 		break;
 
 	case PTM_BFD_ADM_DOWN:
+		stream_putl(msg, BFD_STATUS_ADMIN_DOWN);
```

Comment:
Do you plan to use this state somewhere? I see that you added support to send it, but you've translated it everywhere to `BFD_STATUS_DOWN` again.

Copied from: https://github.com/FRRouting/frr/pull/5002#discussion_r325666606

Reply:
A new state BFD_STATUS_ADMIN_DOWN is sent to all protocols.
Protocols do not take any action related to protocol session if the session is ADMIN_DOWN and only record that BFD session is DOWN. This helps to ensure protocol session is not DOWN/impacted when BFD is ADMIN_DOWN.

The reason for translating ADMIN_DOWN to DOWN in protocol code is, we do not want protocol to be aware of all the states of BFD except UP/DOWN.